### PR TITLE
chore(typespec): bump `typespec` to `0.48.0`

### LINKED ISSUE #3745: [dpg] upgrade to typespec 0.48.0
- Upgrade `typespec` to `0.48.0`
- Upgrade `cadl-ranch` to `0.22.0`
- Upgrade depending libraries

### PR DESCRIPTION
**Pull request by archerzz**
- bump `typespec` to `0.48.0`
- bump `cadl-ranch` to `0.22.0`
- replace `getSdkSimpleType` (which is removed) with `getSdkBuiltInType`
- fix lro tsp definitions due to core library update
- fix errors due to stricter syntax check

resolve #3745

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: samples/AnomalyDetector/client.tsp [hunk 1]
```diff
@@ -3,14 +3,14 @@ import "./main.tsp";
...
 using Azure.ClientGenerator.Core;
 
-@@convenientAPI(AnomalyDetector.Multivariate.GetMultivariateBatchDetectionResult, true)
-@@convenientAPI(AnomalyDetector.Multivariate.TrainMultivariateModel, true)
-@@convenientAPI(AnomalyDetector.Multivariate.ListMultivariateModels, true)
-@@convenientAPI(AnomalyDetector.Multivariate.DeleteMultivariateModel, true)
-@@convenientAPI(AnomalyDetector.Multivariate.GetMultivariateModel, true)
-@@convenientAPI(AnomalyDetector.Multivariate.DetectMultivariateBatchAnomaly, true)
-@@convenientAPI(AnomalyDetector.Multivariate.DetectMultivariateLastAnomaly, true)
+@@convenientAPI(AnomalyDetector.Multivariate.GetMultivariateBatchDetectionResult, true);
```
Comment (pshao25) [inline samples/AnomalyDetector/client.tsp:6]: What if there is no semicolon

Comment (archerzz) [reply samples/AnomalyDetector/client.tsp:6]: > What if there is no semicolon

some compiler errors

**Review by pshao25 — APPROVED** [review]

**Comment by pshao25** [comment]
Looks good to me, but let's have a regen.

File: test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json [hunk 2]
```diff
@@ -589,7 +589,7 @@
      "HttpMethod": "POST",
      "RequestBodyMediaType": "Json",
      "Uri": "{lroBasicUrl}",
-     "Path": "/",
+     "Path": "/thing",
```
Comment (lirenhe) [inline test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json:592]: I didn't find the tsp update for this, so what triggered this change?

Comment (archerzz) [reply test/TestProjects/Lro-Basic-TypeSpec/src/Generated/tspCodeModel.json:592]: there is a new warning asking to add the `@route` for `createThing`. So I added it: https://github.com/Azure/autorest.csharp/pull/3751/files#diff-ba44755667678a56498528ee8f031560d52d6c08902325acfce0fc58f1034794R63

**Comment by archerzz** [comment]
regen preview: https://github.com/Azure/azure-sdk-for-net/pull/38822

**Review by lirenhe — APPROVED** [review]

**Pull request merged**